lib/ui/painting.dart

```diff
@@ -1742,6 +1742,14 @@ enum PathOperation {
   reverseDifference,
 }
 
+/// A handle for the framework to hold and retain an engine layer across frames.
+class EngineLayer extends NativeFieldWrapperClass2 {
+  /// This class is created by the engine, and should not be instantiated
+  /// or extended directly.
+  @pragma('vm:entry-point')
```

Comment:
Just curious about how these pragmas work. Can we decorate the class with the same to indicate that all methods of the same are potential entrypoints?

Reply:
I don't know since I just followed what @alexmarkov suggested. Alex might know the answer?

Reply:
Entry point annotation on class means that its instances can be created from non-Dart code. It does not mark all members as entry points. If C++ code calls particular constructor, then that particular constructor should be marked with pragma.

Also, it makes sense to annotate only those members which are used from non-Dart code, as annotating all members may just increase the size of snapshot and make global analysis less precise.

More about syntax and meaning of entry point pragmas: https://github.com/dart-lang/sdk/blob/master/runtime/vm/compiler/aot/entry_points_pragma.md